lib/reek/smells/smell_warning.rb

```diff
@@ -53,12 +55,17 @@ def matches_smell_type?(klass)
         smell_classes.include?(klass.to_s)
       end
 
-      def matches_smell_details?(other_smell_details = {})
-        common_parameters_equal?(other_smell_details)
+      def matches_smell_details?(other_details = {})
```

Comment:
This method is very confusing because of the naming of the methods we use in there. 
I also think I made an error when I suggested using "_details". I believe we should rather name everything here and below "_attributes", because this is in fact what we're checking for - "attributes", not "details". The wording "details" made sense before but doesn't anymore.

How about doing something like this:

```Ruby
def matches_attributes(other_attributes = {})
  return false unless common_parameters_equal?(other_attributes[:parameters])
  other_attributes.delete(:parameters)
  other_attributes.all? do |other_key, other_value|
    send(other_key) == other_value
  end
end
```

And then use "check_details_exist" only in "common_parameters_equal?" like it was before. 

```Ruby
    send(other_key) == other_value
```

will raise on invalid attributes in any case.

And then we don't need "check_details_exist" at all and you can just put the method body at the beginning of "common_parameters_equal?"